modules/oagw/oagw-gw/src/domain/ports.rs

```diff
@@ -0,0 +1,49 @@
+//! Output ports (interfaces) for domain services.
+
+use async_trait::async_trait;
+use modkit_security::SecurityContext;
+use oagw_sdk::Secret;
+use uuid::Uuid;
+
+use super::error::DomainError;
+
+/// Port for resolving secrets from cred_store.
+#[async_trait]
+pub trait SecretResolver: Send + Sync {
```

Comment:
Do we really need t oabstract secret resolver via ports pattern? I thought that we can call Credstore (whatever) client here.

Anyway ports in domain layer should be traits not concrete implementations. (https://github.com/hypernetix/hyperspot/blob/main/examples/modkit/users_info/users_info/src/domain/ports/audit.rs) 

The specific port implementation should be implemented in src/infra (https://github.com/hypernetix/hyperspot/blob/main/examples/modkit/users_info/users_info/src/infra/audit/http_audit_client.rs)